fw/http_msg.c

```diff
@@ -1525,24 +1525,36 @@ __tfw_h2_msg_shrink_frags(TfwMsgIter *it, const char *body,
 	}
 }
 
+static void
+tfw_h2_body_skb_set(TfwHttpResp *resp, struct sk_buff *nskb)
+{
+	if (test_bit(TFW_HTTP_B_CHUNKED, resp->flags)) {
+		TfwStr *h2_body = __tfw_h2_get_body_start(resp);
+		h2_body->skb = nskb;
```

Comment:
There may be few problems:
 
1.  You replace skb pointer only for "data chunks"(`__tfw_h2_get_body_start` skips "cut chunks"), although you need to replace skb for both "cut" and "data chunks". 
2.  `__tfw_h2_get_body_start` can't determine whether the found body start is in currently transformed skb.
3. `resp->body` must point to new skb anyway, because we will cutoff all cut chunks start from `resp->body`.

You can use `encoding` test suite with low mtu for debugging this part.